### PR TITLE
Core and naming refactor

### PR DESCRIPTION
**Pull request by sanderpick**
Name changes:
- threadstore -> logstore
- eventstore -> store
- threadservice -> service 

The needed pieces from `go-textile-core` have been moved into here and moved around into logical locations.

### CODE REVIEW

File: api/client/read.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 
 	pb "github.com/textileio/go-threads/api/pb"
-	es "github.com/textileio/go-threads/eventstore"
+	"github.com/textileio/go-threads/store"
```
Comment (jsign) [inline api/client/read.go]: Feels much better :)

**Comment by jsign** [comment]
> Name changes:
> 
>     * threadstore -> logstore
> 
>     * eventstore -> store
> 
>     * threadservice -> service
> 
> 
> The needed pieces from `go-textile-core` have been moved into here and moved around into logical locations.

Eventually, `go-textile-core` will be marked as deprecated/archived?

**Comment by jsign** [comment]
Seems that there's an import cycle:
```
go test -v ./...

import cycle not allowed
package github.com/textileio/go-threads/api
	imports github.com/textileio/go-threads/core/service
	imports github.com/textileio/go-threads/core/logstore
	imports github.com/textileio/go-threads/core/service
can't load package: import cycle not allowed
package github.com/textileio/go-threads/api
	imports github.com/textileio/go-threads/core/service
	imports github.com/textileio/go-threads/core/logstore
	imports github.com/textileio/go-threads/core/service

Exited with code exit status 1
```

**Comment by sanderpick** [comment]
Yep, ran out of time on Tuesday... 👀

**Comment by sanderpick** [comment]
I moved `ID`, `LogInfo`, etc back to a `thread` package... feels much better. Thanks compiler 👍

**Comment by sanderpick** [comment]
I just added [DEPRECATED] to https://github.com/textileio/go-textile-core title, and archived it.

**Pull request merged**